packages/api/get-service-stops.ts

```diff
@@ -24,56 +24,63 @@ export const getQueryInput = (event: APIGatewayEvent): ServiceStopsQueryInput =>
-                    localityName: stop.fromLocalityName,
-                    parentLocalityName: stop.fromParentLocalityName,
-                    longitude: stop.fromLongitude,
-                    latitude: stop.fromLatitude,
-                    stopType: stop.fromStopType,
-                    busStopType: stop.fromBusStopType,
-                    timingStatus: stop.fromTimingStatus,
-                    administrativeAreaCode: stop.fromAdministrativeAreaCode,
-                    status: stop.fromStatus,
-                });
-            }
+        if (stop.toStatus === "active") {
+            stopArray.push({
+                id: stop.toId,
+                atcoCode: stop.toAtcoCode,
+                naptanCode: stop.toNaptanCode,
+                commonName: stop.toCommonName,
+                street: stop.toStreet,
+                indicator: stop.toIndicator,
+                bearing: stop.toBearing,
+                nptgLocalityCode: stop.toNptgLocalityCode,
+                localityName: stop.toLocalityName,
+                parentLocalityName: stop.toParentLocalityName,
+                longitude: stop.toLongitude,
+                latitude: stop.toLatitude,
+                stopType: stop.toStopType,
+                busStopType: stop.toBusStopType,
+                timingStatus: stop.toTimingStatus,
+                administrativeAreaCode: stop.toAdministrativeAreaCode,
+                status: stop.toStatus,
+                direction: stop.direction || "",
+                sequenceNumber: stop.fromSequenceNumber || "",
+            });
+        }
 
-            if (stop.toStatus === "active") {
-                stopArray.push({
-                    id: stop.toId,
-                    atcoCode: stop.toAtcoCode,
-                    naptanCode: stop.toNaptanCode,
-                    commonName: stop.toCommonName,
-                    street: stop.toStreet,
-                    indicator: stop.toIndicator,
-                    bearing: stop.toBearing,
-                    nptgLocalityCode: stop.toNptgLocalityCode,
-                    localityName: stop.toLocalityName,
-                    parentLocalityName: stop.toParentLocalityName,
-                    longitude: stop.toLongitude,
-                    latitude: stop.toLatitude,
-                    stopType: stop.toStopType,
-                    busStopType: stop.toBusStopType,
-                    timingStatus: stop.toTimingStatus,
-                    administrativeAreaCode: stop.toAdministrativeAreaCode,
-                    status: stop.toStatus,
-                });
-            }
+        return stopArray;
+    });
+};
 
-            return stopArray;
-        })
-        .filter((stop, index, self) => index === self.findIndex((t) => t.atcoCode === stop.atcoCode));
+export const formatStops = (stops: ServiceStops): Stops => {
+    return flattenStops(stops).filter(
+        (stop, index, self) => index === self.findIndex((t) => t.atcoCode === stop.atcoCode),
```

Comment:
if we're filtering over the flattened stops, it would make sense that the name of the variable would be 'flattenedStop' which means we can use 'stop' instead of 't' for the findIndex.

Reply:
done